File: testdata/bench/go/BenchmarkTest00021.go
package bench

import (
	"net/http"
	"os/exec"
)

// Command injection via exec.Command
func Handler00021(w http.ResponseWriter, r *http.Request) {
	cmd := r.URL.Query().Get("cmd")
	exec.Command("sh", "-c", cmd).Run()
}

File: testdata/bench/rust/BenchmarkTest00021.rs
use actix_web::{web, HttpResponse};
use std::collections::HashMap;
use std::process::Command;

// Vulnerable: Command injection via shell -c
async fn handler(query: web::Query<HashMap<String, String>>) -> HttpResponse {
    let param = query.get("cmd").unwrap_or(&String::new()).clone();
    let output = Command::new("sh").arg("-c").arg(&param).output().expect("failed");
    HttpResponse::Ok().body(String::from_utf8_lossy(&output.stdout).to_string())
}
